sky/skylet/providers/gcp/config.py

```diff
@@ -753,28 +763,53 @@ def _create_rules(config, compute, rules, VPC_NAME, PROJ_ID):
         wait_for_compute_global_operation(config["provider"]["project_id"], op, compute)
 
 
-def get_usable_vpc(config):
+def get_usable_vpc(config) -> str:
     """Return a usable VPC.
 
+    If config['provider']['vpc_name'] is set, return the VPC with the name
+    (errors out if not found). When this field is set, no firewall rules
+    checking or overrides will take place; it is the user's responsibility to
+    properly set up the VPC.
+
     If not found, create a new one with sufficient firewall rules.
+
+    Raises:
+      RuntimeError: if the user has specified a VPC name but the VPC is not found.
     """
     _, _, compute, _ = construct_clients_from_provider_config(config["provider"])
-
-    # For backward compatibility, reuse the VPC if the VM is launched.
-    resource = GCPCompute(
-        compute,
-        config["provider"]["project_id"],
-        config["provider"]["availability_zone"],
-        config["cluster_name"],
-    )
-    node = resource._list_instances(label_filters=None, status_filter=None)
-    if len(node) > 0:
-        netInterfaces = node[0].get("networkInterfaces", [])
-        if len(netInterfaces) > 0:
-            vpc_name = netInterfaces[0]["network"].split("/")[-1]
-            return vpc_name
-
-    vpcnets_all = _list_vpcnets(config, compute)
+    specific_vpc_to_use = config["provider"].get("vpc_name", None)
+    if specific_vpc_to_use is None:
+        # For backward compatibility, reuse the VPC if the VM is launched.
+        resource = GCPCompute(
+            compute,
+            config["provider"]["project_id"],
+            config["provider"]["availability_zone"],
+            config["cluster_name"],
+        )
+        node = resource._list_instances(label_filters=None, status_filter=None)
+        if len(node) > 0:
+            netInterfaces = node[0].get("networkInterfaces", [])
+            if len(netInterfaces) > 0:
+                vpc_name = netInterfaces[0]["network"].split("/")[-1]
+                return vpc_name
+
+        vpcnets_all = _list_vpcnets(config, compute)
+    else:
+        vpcnets_all = _list_vpcnets(
+            config, compute, filter=f"name={specific_vpc_to_use}"
+        )
+        assert (
+            len(vpcnets_all) <= 1
+        ), f"{len(vpcnets_all)} VPCs found with the same name {specific_vpc_to_use}"
```

Comment:
Should this be raised or just assert?

Reply:
It's to catch internal programming error (on GCP, VPC names are unique, so it'd be 0 or 1 VPC found), so assert is appropriate? Added a comment.